classes/class-ledyer-lco-gateway.php

```diff
@@ -242,6 +242,15 @@ public function enqueue_scripts() {
 		public function process_payment( $order_id ) {
 			$order = wc_get_order( $order_id );
 
+			// HPP Redirect flow.
+			if ( 'redirect' === ( $this->settings['checkout_flow'] ?? 'embedded' ) ) {
```

Comment:
We should add the checks for pay for order as well. You can look at other plugins we have made for this if you want, for example [Klarna Checkout](https://github.com/krokedil/klarna-checkout-for-woocommerce/blob/master/classes/class-kco-gateway.php#L219)